powerdns_recursor/datadog_checks/powerdns_recursor/powerdns_recursor.py

```diff
@@ -144,24 +165,31 @@ def _get_config(self, instance):
         return Config(host, port, version), tags
 
     def _get_pdns_stats(self, config, tags):
-        fallback_url = "http://{}:{}/api/v1/servers/localhost/statistics".format(config.host, config.port)
-        if config.version == 4:
-            url = fallback_url
-        else:
-            url = "http://{}:{}/servers/localhost/statistics".format(config.host, config.port)
+        url_v4 = "http://{}:{}/api/v1/servers/localhost/statistics".format(config.host, config.port)
+        url = "http://{}:{}/servers/localhost/statistics".format(config.host, config.port)
 
         service_check_tags = ['recursor_host:{}'.format(config.host), 'recursor_port:{}'.format(config.port)] + tags
+
+        try:
+            request = self._get_pdns_response(config, url, url_v4)
+        except Exception:
+            self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.CRITICAL, tags=service_check_tags)
+            raise
+
+        self.service_check(self.SERVICE_CHECK_NAME, AgentCheck.OK, tags=service_check_tags)
+        return request.json()
+
+    def _get_pdns_response(self, config, url, url_v4):
+        if config.version == 4:
+            url = url_v4
+
         try:
             request = self.http.get(url)
             request.raise_for_status()
```

Comment:
I know it's not strictly in the scope of this PR bc it was like that before, but this variable is a `response`:

```suggestion
            response = self.http.get(url)
            response.raise_for_status()
```